src/api.rs

```diff
@@ -78,36 +82,29 @@ pub struct FeedPlayerResponse {
 #[post("/feed-me", format = "application/json", data = "<payload>")]
 pub fn feed_player(
     payload: JSON<FeedPlayerRequest>,
-    scoreboard: State<Mutex<Scoreboard>>,
+    players: State<PlayerMap>,
     broadcaster: State<HostBroadcaster>,
 ) -> Result<JSON<FeedPlayerResponse>>
 {
     let payload = payload.into_inner();
-    let player_id = payload.player;
+    let id = payload.player;
 
     // Add 1 to the player's score, returning the new score.
-    let score = {
-        let mut scoreboard = scoreboard.lock().expect("Scoreboard mutex was poisoned");
+    let balls = {
+        let mut players = players.write().expect("Players were poisoned");
 
         // Get the player's current score, or return an `InvalidPlayer` error if it's not in
         // the scoreboard.
-        let score = scoreboard
-            .get_mut(&player_id)
-            .ok_or(Error::InvalidPlayer(player_id))?;
-        *score += 1;
-        *score
```

Comment:
Ooo, a `safe`/`unsafe` mode is a good idea. Allows both the dangerous superpowers and the reasonable limitations.